ee/tabby-webserver/src/service/background_job/mod.rs

```diff
@@ -101,14 +138,16 @@ pub async fn start(
                         continue;
                     }
 
-                    let logger = JobLogger::new(db.clone(), job.id);
-                    debug!("Background job {} started, command: {}", job.id, job.command);
+                    let job_id = job.id;
+                    let logger = JobLogger::new(db.clone(), job_id);
+                    debug!("Background job {} started, command: {}", job_id, job.command);
                     let Ok(event) = serde_json::from_str::<BackgroundJobEvent>(&job.command) else {
                         logkit::info!(exit_code = -1; "Failed to parse background job event, marking it as failed");
                         continue;
                     };
 
-                    if let Err(err) = match event {
+                    let event_clone = event.clone();
```

Comment:
use match &event to avoid the clone?

Reply:
if we do not clone event, then we have to clone in L151, L155, L159 and L163 for the members of event